File: go/difference-of-squares/difference_of_squares.go
package diffsquares

import "math"

func SquareOfSum(n int) int {
	var s = 0

	for i := 1; i <= n; i++ {

		s += i

	}

	return s * s

}

func Difference(n int) int {

	return SquareOfSum(n) - SumOfSquares(n)

}

func SumOfSquares(n int) int {

	var s2 = 0.0

	for i := 1; i <= n; i++ {

		s2 += math.Pow(float64(i), 2)

	}

	return int(s2)
}

File: nim/difference-of-squares/difference_of_squares.nim
import math, sequtils

func squareOfSum*(n: int): int = toSeq(1..n).sum ^ 2
func sumOfSquares*(n: int): int = toSeq(1..n).mapIt(it^2).sum
func difference*(n: int): int = squareOfSum(n) - sumOfSquares(n)
